### ISSUE TITLE
The training results are poor

### ISSUE THREAD

**Issue by EverlastingYeay** [RicardooYoung/VersatileTeacher#4]
Hello! I encountered some problems when using the VT model for training and the results were not satisfactory. After training the teacher model and the student model with the cityscapes - cityscapes_foggy(0.02) dataset, the mAP_0.5 results I obtained were 0.329 and 0.263 respectively, which is much lower than the 0.515 effect mentioned in the paper. I sincerely look forward to your reply. Thank you!

**Comment by RicardooYoung** [comment]
It seems that there is an issue with the initialization of both models. Before training VT, pretrained weights should be acquired by running the following script:
```angular2html
python train.py \
    --cfg models/yolov5l_da.yaml \
    --weights weights/yolov5l.pt \
    --data data/cityscapes.yaml \
    --data data/cityscapes_foggy.yaml \
    --device 0 \
    --epochs 50 \
    --batch-size 16
```
Then, run the following script:
```angular2html
python train.py \
    --cfg models/yolov5l_da.yaml \
    --cfg models/yolov5l.yaml \
    --weights weights/yolov5l_cityscapes.pt \
    --weights weights/yolov5l_cityscapes.pt \
    --data data/cityscapes.yaml \
    --data data/cityscapes_foggy.yaml \
    --device 0 \
    --epochs 50 \
    --batch-size 16
```
using the weight ``yolov5l_cityscapes.pt`` obtained in the previous step.

Hope this helps you!

**Comment by EverlastingYeay** [comment]
> It seems that there is an issue with the initialization of both models. Before training VT, pretrained weights should be acquired by running the following script:
> 
> ```
> python train.py \
>     --cfg models/yolov5l_da.yaml \
>     --weights weights/yolov5l.pt \
>     --data data/cityscapes.yaml \
>     --data data/cityscapes_foggy.yaml \
>     --device 0 \
>     --epochs 50 \
>     --batch-size 16
> ```
> 
> Then, run the following script:
> 
> ```
> python train.py \
>     --cfg models/yolov5l_da.yaml \
>     --cfg models/yolov5l.yaml \
>     --weights weights/yolov5l_cityscapes.pt \
>     --weights weights/yolov5l_cityscapes.pt \
>     --data data/cityscapes.yaml \
>     --data data/cityscapes_foggy.yaml \
>     --device 0 \
>     --epochs 50 \
>     --batch-size 16
> ```
> 
> using the weight `yolov5l_cityscapes.pt` obtained in the previous step.
> 
> Hope this helps you!

Thank you for your reply. However, I trained in accordance with what you mentioned above previously. I followed the steps in the readme to train the teacher model first, and then the student model.

**Comment by RicardooYoung** [comment]
I’m not exactly sure why this happens. You might try reducing the learning rate during the second training phase to see if it leads to any improvement.

**Comment by EverlastingYeay** [comment]
> I’m not exactly sure why this happens. You might try reducing the learning rate during the second training phase to see if it leads to any improvement.


Thank you for your suggestion, it indeed worked. Now, after training the teacher model and the student model with the cityscapes - cityscapes_foggy(0.02) dataset, the mAP_0.5 results I obtained were 0.387 and 0.34, respectively. However, the performance is still quite far from the results reported in the paper. Additionally, I noticed that the performance of the student model continues to decline during training, dropping from an initial 40.2% to 34.1%.I will attach the result.csv file from the training。
[results.csv](https://github.com/user-attachments/files/18305462/results.csv)

I would like to confirm whether the two results output after each epoch correspond to the training set and the validation set, respectively. I have been using the second result for evaluation. Could you please advise if there might be any issues causing this?

Thank you again for your previous  responses, and I look forward to hearing from you this time as well.

**Comment by RicardooYoung** [comment]
The output results after each epoch represent the source domain and the target domain, both on the validation set.

In my experience, the performance decline may be due to the following reasons:

- Poor pretraining
- The learning rate is too high
- An improper initial threshold

You can try adjusting these parameters to see if it improves the performance.

**Comment by momo123321967** [comment]
Hello, can I ask about your step1 training results? My step1 training results are very poor.

**Comment by EverlastingYeay** [comment]
> Hello, can I ask about your step1 training results? My step1 training results are very poor.
hello,on cityscapes - cityscapes_foggy(0.02) dataset, the mAP_0.5 results I obtained after step 1 were 0.387

**Comment by momo123321967** [comment]
thank you so much!

**Comment by momo123321967** [comment]
I used yolov5s for step-1 training, and the mAP50 of the step-1 training were 0.376(on cs) and 0.325(on fcs). But the results of step-2 training were very poor (using hyp.yaml provided by the author), so I began to doubt whether this teacher-student self-training framework is really useful? 
It seems that current domain adaptation target detection all adopts this framework.

**Comment by EverlastingYeay** [comment]
> The output results after each epoch represent the source domain and the target domain, both on the validation set.
> 
> In my experience, the performance decline may be due to the following reasons:
> 
> * Poor pretraining
> * The learning rate is too high
> * An improper initial threshold
> 
> You can try adjusting these parameters to see if it improves the performance.

Thank you for your suggestions. I have actually tried adjusting the parameters you mentioned multiple times, such as those related to pretraining, learning rate, and the initial threshold. However, unfortunately, the performance still hasn't shown significant improvement.
I am wondering if there is a possibility that you could send me the model weights of the pretraining, specifically the "weights/yolov5l_teacher.pt file". I suspect that having access to this particular pretrained model weights might help me to better address the performance issue I am facing. It could potentially provide a more solid foundation for my subsequent training and might lead to better results.
Thank you very much for your time and consideration. I am looking forward to your reply.

**Comment by adsadswdas** [comment]
Hello, have you successfully reproduced the data presented in the paper？
